mesonbuild/options.py

```diff
@@ -1128,6 +1128,11 @@ def is_project_option(self, key: OptionKey) -> bool:
         """Convenience method to check if this is a project option."""
         return key in self.project_options
 
+    def is_per_machine_option(self, optname: OptionKey) -> bool:
+        if optname.evolve(subproject=None, machine=MachineChoice.HOST) in BUILTIN_OPTIONS_PER_MACHINE:
```

Comment:
I was trying to understand why we evolve this to subproject=None, though maybe it's a bugfix for the option rewrite?

Reply:
It likely is a bug fix. We need to evolve this because the keys in the `BUILTIN_OPTIONS_PER_MACHINE` as always at the global scope for the host, so they look something like:
```python
{OptionKey('pkg_config_path', None, MachineChoice.HOST): ...}
```
But we can get subproject overridden or build machine keys here, which should match but don't, so if you got `OptionKey('pkg_config_path', None, MachineChoice.BUUILD)`, we expect that to work, but it wouldn't.

It's also quite possible this is a bug that theoretically could exist, but doesn't get hit currently.